lmod_proxy/edx_grades/__init__.py

```diff
@@ -14,13 +24,51 @@
 )
 
 
-@edx_grades.route('/', methods=['GET'])
+@edx_grades.route('', methods=['GET', 'POST'])
 def index():
     """Handle ``POST`` from edx-platform or print available actions, and
     provide an interactive test of the available actions on ``GET``.
 
     Returns:
-        flask.response
-
+        flask.response: JSON on POST, HTML on GET
     """
-    return '<h1>LMod Proxy edX Grades "API"</h1>'
+    requestor = request.remote_addr
+    if request.headers.getlist("X-Forwarded-For"):
+        requestor = request.headers.getlist("X-Forwarded-For")[0]
+
+    if request.method == 'POST':
+        form = EdXGradesForm(request.form)
```

Comment:
Just add to this, the "API" from edX is a form. You'll notice the API: https://github.com/edx/edx-platform/blob/master/docs/en_us/internal/remote_gradebook.md#for-submitpost-grades just changes the value of the submit button to change the action.  So form is what we are getting, so form is how we are processing.

All that said, I still think using WTForms and doing form validation of API even if it was JSON instead of a form is still a very clean way to validate the data.  It handles cleanup and validation of any expected data without complicated if-else-elseif.  i.e. you just do `MyForm(request.json()).validate()` and you know everything meets your expectation.